src/main/webapp/script.js

```diff
@@ -220,51 +220,117 @@ function getGraphDisplay(graphNodes, graphEdges) {
     } else {
-      document.getElementById('search-error').innerText = "Node does not exist.";
+      errorText = type + " does not exist.";
     }
-  };
-  
+  }
+  document.getElementById(type + '-error').innerText = errorText;
+  return result;
 }
 
 /**
- * Zooms in on specific node
+ * Finds specific node and zooms in
  */
 function searchNode(cy, query) {
-  // reset nodes to default color
+  if (query) {
+    const target = cy.$id(query);
+    if (target.length != 0) {
+      highlightElements(cy, target);
+      cy.fit(target, 50);
+      return target;
+    }
+  }
+}
+/**
+ * Constructs list of nodes that contain specified token
+ * and zooms in
+ */
+function searchToken(cy, query) {
+  let target = cy.collection();
   cy.nodes().forEach(node => {
-    node.style('background-color', 'blue');
-    node.style('opacity', '1')
+    if (node.data().tokens.includes(query)) {
+      target = target.add(node);
+    }
   });
-  const target = findNodeInGraph(cy, query);
-  if (target) {
-    cy.nodes().forEach(node => node.style('opacity', '0.25'));
-    target.style('background-color', 'olive');
-    target.style('opacity', '1');
-    cy.fit(target, 50);
-    return true;
-  } else {
-    // fits all nodes on screen
-    cy.fit(cy.nodes(), 50);
-    return false;
+  if (target.length > 0) {
+    let showNode = target[0][0];
+    showNode.tip.show();
+    highlightElements(cy, target);
+    return target;
   }
 }
 
 /**
- * Finds element in cy graph by id
+ * Highlights collection of nodes and edges
  */
-function findNodeInGraph(cy, id) {
-  if (id.length != 0) {
-    const target = cy.$('#'+id);
-    if (target.length != 0) {
-      return target;
+function highlightElements(cy, target) {
+  // set all nodes to background state
+  // set to 0.24 to distinguish between being blurred
+  // for highlighting or because of a mutation
+  cy.nodes().forEach(node => node.style('opacity', '0.24'));
+
+  // highlight desired nodes
+  target.forEach(node => {
+    node.style('border-width', '4px');
+    node.style('opacity', '1');
+  });
+  cy.fit(target[0], 50);
+  document.getElementById('num-selected').innerText = "Number of nodes selected: " + target.length;
+
+  // highlight adjacent edges
+  target.connectedEdges().forEach(edge => {
+    edge.style('line-color', 'black');
+    edge.style('target-arrow-color', 'black');
+    edge.style('z-index', '2');
+  });
+}
+
+/**
+ * Resets highlighted elements to default state
+ */
+function resetElements(cy) {
+  // reset node borders and opacity
+  cy.nodes().forEach(node => {
+    node.style('border-width', '0px');
+    // only change opacity of nodes that were changed
+    // because of highlighting
+    if (node.style('opacity') == "0.24") {
+      node.style('opacity', '1');
     }
-  }
-  return null;
+  });
+
+  // reset edge color
+  cy.edges().forEach(edge => {
+    // only change color of edges that were changed
+    // because of highlighting
+    if (edge.style('line-color') == 'rgb(0,0,0)') {
```

Comment:
is there a reason you define the color with rgb here and `#000` in other places?

Reply:
also `===`